test/e2e/emitters/kgateway/testdata/output/ssl_redirect.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  annotations:
+    gateway.networking.k8s.io/generator: ingress2gateway-dev
+  name: nginx
+  namespace: default
+spec:
+  gatewayClassName: kgateway
+  listeners:
+  - hostname: ssl-redirect.localdev.me
+    name: ssl-redirect-localdev-me-http
+    port: 80
+    protocol: HTTP
+  - hostname: ssl-redirect.localdev.me
+    name: ssl-redirect-localdev-me-https
+    port: 443
+    protocol: HTTPS
+    tls:
+      certificateRefs:
+      - group: null
+        kind: null
+        name: ssl-redirect-tls
+status: {}
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  annotations:
+    gateway.networking.k8s.io/generator: ingress2gateway-dev
+  name: ssl-redirect-localhost-ssl-redirect-localdev-me-https
+  namespace: default
+spec:
+  hostnames:
+  - ssl-redirect.localdev.me
+  parentRefs:
+  - name: nginx
+    sectionName: ssl-redirect-localdev-me-https
+  rules:
+  - backendRefs:
+    - name: echo-backend
+      port: 8080
+    matches:
+    - path:
+        type: PathPrefix
+        value: /
+status:
+  parents: []
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  annotations:
+    gateway.networking.k8s.io/generator: ingress2gateway-dev
+  name: ssl-redirect-localhost-ssl-redirect-localdev-me-http-redirect
+  namespace: default
+spec:
+  hostnames:
+  - ssl-redirect.localdev.me
+  parentRefs:
+  - name: nginx
+    sectionName: ssl-redirect-localdev-me-http
+  rules:
+  - filters:
+    - requestRedirect:
+        scheme: https
+        statusCode: 301
```

Comment:
See https://github.com/kgateway-dev/ingress2gateway/pull/23/files#r2627873617 (301 can be updated to 308 throughout).